apps/web/src/pages/templates/editor/TemplateEditorPage.tsx

```diff
@@ -1,16 +1,29 @@
 import { FormProvider } from 'react-hook-form';
-import { Group, Container } from '@mantine/core';
+import { Container, Group } from '@mantine/core';
+import { useState } from 'react';
+import { ChannelTypeEnum } from '@notifire/shared';
 import { useParams } from 'react-router-dom';
 import PageContainer from '../../../components/layout/components/PageContainer';
 import PageHeader from '../../../components/layout/components/PageHeader';
 import { TemplatesSideBar } from '../../../components/templates/TemplatesSideBar';
 import { NotificationSettingsForm } from '../../../components/templates/NotificationSettingsForm';
 import { useTemplateController } from '../../../legacy/pages/templates/editor/use-template-controller.hook';
 import { TemplateTriggerModal } from '../../../components/templates/TemplateTriggerModal';
+import { AddChannelsPage } from './AddChannelsPage';
 import { Button } from '../../../design-system';
 
 export default function TemplateEditorPage() {
   const { templateId = '' } = useParams<{ templateId: string }>();
+  const [activePage, setActivePage] = useState<string>('Settings');
+  const [channelButtons, setChannelButtons] = useState<string[]>([]);
+
+  const handleAddChannel = (tabKey) => {
+    const index = channelButtons.findIndex((item) => item === tabKey);
+    if (index === -1) {
```

Comment:
```suggestion
    const foundChannel = channelButtons.find((item) => item === tabKey);
    if (foundChannel) {
```

A small suggestion, I think semantically it's better to use find here, don't see a reason to use index search if we are not using this